internal/transport/transport_test.go

```diff
@@ -2450,3 +2452,20 @@ func TestConnectionError_Unwrap(t *testing.T) {
 		t.Error("ConnectionError does not unwrap")
 	}
 }
+
+// Verify Peer is set in server context.
+func TestPeerSetInServerContext(t *testing.T) {
```

Comment:
Please specify our test struct as the receiver for this test. This will ensure that our leakcheck runs at the end of the test.

func (s) TestPeerSetInServerContext(t *testing.T) { ... }